app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ReplyForwardHeaderManager.kt

```diff
@@ -0,0 +1,183 @@
+            MessageBodyUtils.mergeSplitBodyAndSubBodies(previousBody, body.subBodies, message.uid)
+        } ?: previousBody
+    }
+
+    private fun Recipient.quotedDisplay(): String = "${("$name ").ifBlank { "" }}<$email>"
+
+    private fun assembleForwardHtmlHeader(
+        message: Message,
+        previousFullBody: String
+    ): String = with(appContext) {
+        val messageForwardHeader = getString(R.string.messageForwardHeader)
+        val fromTitle = getString(R.string.fromTitle)
+        val dateTitle = getString(R.string.dateTitle)
+        val subjectTitle = getString(R.string.subjectTitle)
+        val toTitle = getString(R.string.toTitle)
+        val ccTitle = getString(R.string.ccTitle)
+
+        val forwardRoot = "<div class=\"${MessageBodyUtils.INFOMANIAK_FORWARD_QUOTE_HTML_CLASS_NAME}\" />"
+        return newDocumentWithParentElement(forwardRoot).apply {
+            addAndEscapeTextLine("---------- $messageForwardHeader ---------")
+            addAndEscapeTextLine("$fromTitle ${message.fromName()}")
+            addAndEscapeTextLine("$dateTitle ${message.date.toDate()}")
+            addAndEscapeTextLine("$subjectTitle ${message.subject}")
+            addAndEscapeRecipientLine(toTitle, message.to)
+            addAndEscapeRecipientLine(ccTitle, message.cc)
+            addAndEscapeTextLine("")
+            addAndEscapeTextLine("")
+
+            addAlreadyEscapedBody(previousFullBody)
+        }.outerHtml()
+    }
+
+    private fun assembleReplyHtmlHeader(messageReplyHeader: String, previousFullBody: String): String {
+        val replyRoot = """<div id="answerContentMessage" class="${MessageBodyUtils.INFOMANIAK_REPLY_QUOTE_HTML_CLASS_NAME}" />"""
+        return newDocumentWithParentElement(replyRoot).apply {
+            addAndEscapeTextLine(messageReplyHeader, endWithBr = false)
+            addReplyBlockQuote {
+                addAlreadyEscapedBody(previousFullBody)
+            }
+        }.outerHtml()
+    }
+
+    private fun newDocumentWithParentElement(parentElementHtml: String): Element {
+        val doc = Jsoup.parseBodyFragment(parentElementHtml)
+        return doc.body().firstElementChild()!!
+    }
+
+    private fun Element.addAndEscapeTextLine(content: String, endWithBr: Boolean = true) {
+        appendElement("div").apply {
+            text(content)
+            if (endWithBr) appendElement("br")
+        }
+    }
+
+    private fun Element.addAndEscapeRecipientLine(prefix: String, recipientList: List<Recipient>) {
+        formatRecipientList(recipientList)?.let { recipients -> addAndEscapeTextLine("$prefix $recipients") }
+    }
+
+    private fun Element.addAlreadyEscapedBody(previousFullBody: String) {
+        append(previousFullBody)
+    }
+
+    private fun Element.addReplyBlockQuote(addInnerElements: Element.() -> Unit) {
+        val blockQuote = appendElement("blockquote").addClass("ws-ng-quote")
+        blockQuote.addInnerElements()
+    }
```

Comment:
What about making these functions returns the new elements / themselves so we can chain calls ?

Reply:
I don't know if we want to chain them. `apply` already kinda gives us the same way of writing things like "one row after the other"

Reply:
I understand, I just felt that it would have followed the "native way" of creating html

Reply:
Honestly the "native way" is only due to it being Java code